src/OIDC/Crypt.purs

```diff
@@ -49,33 +99,113 @@ foreign import
     -> Maybe StateString
 
 foreign import
+  verifyAudience
+    :: ClientID
+    -> Payload
+    -> Boolean
+
+foreign import
   _pluckKeyId
     :: forall a
      . Maybe a
     -> (a -> Maybe a)
-    -> IdToken
+    -> Header
     -> Maybe KeyId
 
 foreign import
   _pluckEmail
     :: forall a
      . Maybe a
     -> (a -> Maybe a)
-    -> IdToken
+    -> Payload
     -> Maybe Email
 
+foreign import
+  _pluckIAT
+    :: forall a
+     . Maybe a
+    -> (a -> Maybe a)
+    -> Payload
+    -> Maybe Int
+
+foreign import
+  _pluckExp
+    :: forall a
+     . Maybe a
+    -> (a -> Maybe a)
+    -> Payload
+    -> Maybe Int
+
 unbindState
   :: BoundStateJWS
   -> KeyString
   -> Maybe StateString
 unbindState = _unbindState Nothing Just
 
 pluckKeyId
-  :: IdToken
+  :: Header
   -> Maybe KeyId
 pluckKeyId = _pluckKeyId Nothing Just
 
 pluckEmail
-  :: IdToken
+  :: Payload
   -> Maybe Email
 pluckEmail = _pluckEmail Nothing Just
+
+pluckIAT
+  :: Payload
+  -> Maybe Int
+pluckIAT = _pluckIAT Nothing Just
+
+pluckExp
+  :: Payload
+  -> Maybe Int
+pluckExp = _pluckExp Nothing Just
+
+verifyIAT :: Seconds -> Payload -> Boolean
+verifyIAT unixTimeNow payload =
+  maybe false (_ <= unixTimeNow) (Seconds <<< Int.toNumber <$> pluckIAT payload)
+
+verifyExp :: Seconds -> Payload -> Boolean
+verifyExp unixTimeNow payload =
+  maybe false (_ > unixTimeNow) (Seconds <<< Int.toNumber <$> pluckExp payload)
+
+instantToSeconds ∷ Instant -> Seconds
+instantToSeconds = Duration.convertDuration <<< Instant.unInstant
+
+verifyIdToken
+  :: forall eff
+   . Seconds
+  -> IdToken
+  -> Issuer
+  -> ClientID
+  -> UnhashedNonce
+  -> J.JSONWebKey
+  -> Eff (now :: NOW | eff) (Either Exception.Error Boolean)
```

Comment:
We could but in most use cases (including ours) this will only be used as an argument to a print function to get a string again. I can do this if you still would like it.